pkg/api/adjustment.go

```diff
@@ -147,6 +147,20 @@ func (a *ContainerAdjustment) AddCDIDevice(d *CDIDevice) {
 	a.CDIDevices = append(a.CDIDevices, d) // TODO: should we dup d here ?
 }
 
+// AddOrReplaceNamespace records the addition or replacement of the given namespace to a container.
+func (a *ContainerAdjustment) AddOrReplaceNamespace(n *LinuxNamespace) {
+	a.initLinuxNamespaces()
+	a.Linux.Namespaces = append(a.Linux.Namespaces, n) // TODO: should we dup n here ?
```

Comment:
What do you mean by this TODO?  oci only allows one namespace per type and the func we are calling AddOrReplaceLinuxNamespace ensures that..

Reply:
@mikebrow It's related to the similar comments as about mounts and [CDI] devices (as this helper is done consistently with those).

Basically we append here the user-supplied `n *LinuxNamespace` as such to the set of adjustments, as opposed to appending a (deep) copy of it. That, in principle, allows the user to modify it once it has been recorded as an adjustment.

Since the plugin can only shoot itself in the foot with that, I think no too big harm is done. But I think it's good to have the comments in place, so that if we ever decide to change one, we update all the others consistently.